services/frontend-v3/src/components/diplomaTable/DiplomaTable.jsx

```diff
@@ -0,0 +1,172 @@
+import React, { useState, useEffect } from "react";
+import DiplomaTableView from "./DiplomaTableView";
+import { Skeleton } from "antd";
+import axios from "axios";
+import _ from "lodash";
+import { injectIntl } from "react-intl";
+import config from "../../config";
+
+const backendAddress = config.backendAddress;
+
+function DiplomaTable(props) {
+  const [data, setData] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [reset, setReset] = useState(false);
+  const [searchText, setSearchText] = useState("");
+  const [searchedColumn, setSearchedColumn] = useState("");
+  const [selectedRowKeys, setSelectedRowKeys] = useState([]);
+  const [size] = useState("large");
+
+  const { type } = props;
+
+  useEffect(() => {
+    let diplomas = [];
+    if (loading) {
+      const setState = async () => {
+        diplomas = await getDiplomas();
+        setData(diplomas);
+        setLoading(false);
+      };
+      setState();
+    } else {
+      const updateState = async () => {
+        diplomas = await getDiplomas();
+        setData(diplomas);
+        setReset(false);
+      };
+      updateState();
+    }
+  }, [loading, reset]);
+
+  const getDiplomas = async () => {
+    try {
+      let results = await axios.get(
+        backendAddress + "api/admin/options/" + type
+      );
+      return results.data;
+    } catch (error) {
+      console.log(error);
+      return 0;
+    }
+  };
+
+  const getDisplayType = (plural) => {
+    if (plural)
+      return props.intl.formatMessage({
+        id: "admin." + type + ".plural",
+        defaultMessage: type,
+      });
+
+    return props.intl.formatMessage({
+      id: "admin." + type + ".singular",
+      defaultMessage: type,
+    });
+  };
+
+  const handleSearch = (selectedKeys, confirm, dataIndex) => {
+    confirm();
+    setSearchText(selectedKeys[0]);
+    setSearchedColumn(dataIndex);
+  };
+
+  const handleReset = (clearFilters) => {
+    clearFilters();
+    setSearchText("");
```

Comment:
you can just set the field value rather then use a state

Reply:
Resolved.